pip/src/interpreter.rs

```diff
@@ -138,6 +139,9 @@ impl Interpreter {
                 list_directory.expect(
                     "file system hooks should have been passed in with a manifest descriptor",
                 ),
+                resolve_path.expect(
```

Comment:
Perhaps these three things that all need to be passed together could be a tuple or struct?

Reply:
I tried to figure out how to make the tuple - or better yet, named parameters - work across the Python interop boundary here, but got lost pretty quickly. I think that'll have to happen another time, maybe when we touch this code again soon for packages.

For the time being, I at least got rid of the triplicated `expect`s by making the pattern matching a little nicer.